pkg/controller/infrastructure/actuator.go

```diff
@@ -294,6 +292,41 @@ func (a *actuator) isWhitelistedImageID(imageID string) bool {
 	return false
 }
 
+// ensureOldSSHKeyDetached ensures the compatibility when ssh key is generated in the current cluster via terraform.
+func (a *actuator) ensureOldSSHKeyDetached(ctx context.Context, infra *extensionsv1alpha1.Infrastructure) error {
+	if infra.Status.ProviderStatus == nil {
+		return nil
+	}
+
+	infrastructureStatus := &alicloudv1alpha1.InfrastructureStatus{}
+	if _, _, err := a.Decoder().Decode(infra.Status.ProviderStatus.Raw, nil, infrastructureStatus); err != nil {
+		return err
+	}
+
+	// nolint
+	if infrastructureStatus.KeyPairName == "" {
+		return nil
+	}
+
+	_, shootCloudProviderCredentials, err := a.getConfigAndCredentialsForInfra(ctx, infra)
+	if err != nil {
+		return err
+	}
+
+	shootAlicloudECSClient, err := a.newClientFactory.NewECSClient(infra.Spec.Region, shootCloudProviderCredentials.AccessKeyID, shootCloudProviderCredentials.AccessKeySecret)
+	if err != nil {
+		return err
+	}
+	// nolint
+	a.logger.V(2).Info("Detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
+	// nolint
+	err = shootAlicloudECSClient.DetachECSInstancesFromSSHKeyPair(infrastructureStatus.KeyPairName)
+	// nolint:static
```

Comment:
I think we better ignore this file as well in `.golangci.yaml` and remove the `nolint`s in this func.

Locally I get:

```
$ make check
> Check
Executing golangci-lint
WARN [runner/nolint] Found unknown linters in //nolint directives: static
pkg/controller/infrastructure/actuator.go:325:86: SA1019: infrastructureStatus.KeyPairName is deprecated: the field should not be used anymore as the SSH key pair is no longer generated by terraform. (staticcheck)
	a.logger.V(2).Info("Finished detaching ssh key pair from ECS instances", "keypair", infrastructureStatus.KeyPairName)
	                                                                                    ^
make: *** [check] Error 1
```

I don't know why the verify step does not fail with the same.

Reply:
I will remove `static`. The purpose for Deprecated is suggest people not to use it. If we ignore this file, and people may also use this field freely. This is not desired.